### PR TITLE
Add wasi-testsuite to CI

### PR DESCRIPTION
**Pull request by harry900831**
This should be merge after fixing all the issues inside #2249.

I'm new to GitHub Action. If there's any suggestion, please let me know.

Thanks!

### CODE REVIEW

File: .github/workflows/wasi-testsuite.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  test_on_macos:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          path: WasmEdge
+
+      - name: Install dependencies for WasmEdge
+        run: |
+          brew install llvm@14 ninja boost cmake
```
Comment (hydai) [inline .github/workflows/wasi-testsuite.yml:71]: Ditto. You may also try `llvm` instead of `llvm@14`.

**Review by hydai — CHANGES_REQUESTED** [review]

**Review by hydai — DISMISSED** [review]

File: .github/workflows/wasi-testsuite.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      - "thirdparty/wasi/**"
+  pull_request:
+    paths:
+      - ".github/workflows/wasi-testsuite.yml"
+      - "lib/host/wasi/**"
+      - "include/host/wasi/**"
+      - "thirdparty/wasi/**"
+
+jobs:
+  test_on_ubuntu:
```
Comment (loganek) [inline .github/workflows/wasi-testsuite.yml:22]: `test_on_ubuntu` and `test_on_macos` look very similar so I wonder if it makes sense to use [matrix strategies](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) and [if expressions](https://docs.github.com/en/actions/learn-github-actions/expressions) to remove duplications in the job definitions?

**Comment by loganek** [comment]
Hi, is there any plan to merge this PR? Currently the daily runtime validation job fails for WasmEdge due to lack of `.github/scripts/wasi-testsuite-adapter.py` file: https://github.com/WebAssembly/wasi-testsuite/actions/runs/4642264910/jobs/8215987344. I can temporarily disable WasmEdge job but if the PR is ready (or almost ready) to be merged, we should probably do that instead.

File: .github/workflows/wasi-testsuite.yml (hunk 2 above)
Comment (harry900831) [reply .github/workflows/wasi-testsuite.yml]: Thanks for your advise!

**Review by hydai — APPROVED** [review]

**Pull request merged**